File: Leetcode/Python3/455. Assign Cookies/solution.py
# https://leetcode.com/problems/assign-cookies/


class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        result = 0
        size_index = 0
        # Explore small greed children by giving them small cookies.
        for greed in g:
            if size_index == len(s):
                break
            for idx in range(size_index, len(s)):
                if greed <= s[idx]:
                    result += 1
                    size_index = idx + 1
                    break
        return result

File: Leetcode/Typescript/455. Assign Cookies/solution.ts
// https://leetcode.com/problems/assign-cookies/description/

function findContentChildren(g: number[], s: number[]): number {
  g.sort((a, b) => a - b);
  s.sort((a, b) => a - b);
  let pointG = 0;
  let assign = 0;

  for (let idx = 0; idx < s.length; idx++) {
    if (pointG === g.length) {
      break;
    }

    if (s[idx] >= g[pointG]) {
      assign += 1;
      pointG += 1;
    }
  }

  return assign;
}
